### DISCUSSION TITLE
How can I add gaps in TilingSprite? Is this possible?

### DISCUSSION THREAD

**Discussion by spd789562** [pixijs/pixijs#7181] [Q&A]
Is there any way to add some gaps in TilingSprite?
Like Spacing or Overlapping sprite?

**Comment by jonlepage** [comment]
gsap is based on js {object} , so you can handle any object reference you want.
I dont know pixi tillingSprite plugin but try look on `clampMargin`  maybe ? 
https://pixijs.download/dev/docs/PIXI.TilingSprite.html

or based on doc, i dont think TilingSprite have a props for margin !

```js
const sprite = new PIXI.TilingSprite()
gsap.to(sprite, duration, { clampMargin:10} );
```

**Reply by spd789562** [reply]
Last time I used `clampMargin`, it's seems not thing different.
So I used Graphics.drawRect and add Sprite to make a fake gap, but it can't use on negative gap

**Reply by jonlepage** [reply]
hum, try see with the contributor of this plugins, not sure who is the official guy who made this plugins?
May more luck if you poke the creator ! and ask for add this features.
![image](https://user-images.githubusercontent.com/24865815/105257483-de02b800-5b55-11eb-917a-68fe6c7f2222.png)

edit:
ha is @ivanpopelyshev 
![image](https://user-images.githubusercontent.com/24865815/105257923-c0821e00-5b56-11eb-9306-d6fe5c62c098.png)

Good luck it is overloaded with requests, but you can still try open an issue ! 
On my side i think the feature your ask are not avaible for pixi-tile !
`clampMargin` i think is related to displayObject, and not the tile renderer !
So it will only give you marge from renderer and not between tiles.

**Reply by spd789562** [reply]
ok, thanks for responding! I'll try to open an issue for a feature request.

**Reply by jonlepage** [reply]
ho sorry , when i read `gaps` i read `gsap` 😂 oups, wrong technologies , and wrong answer !

**Comment by spd789562** [comment]
Ok, according to #7189, TilingSprite doesn't support gap/margin and not gonna add this feature.
If need positive margin, use function below to create new texture that including space.
```js
function generateGapTexture(texture, gap) {
  const gapBox = new Pixi.Graphics()
  const originSprite = new Pixi.Sprite(texture)
  gapBox.drawRect(0, 0, originSprite.width + gap.x, originSprite.height + gap.y)
  gapBox.addChild(originSprite)
  return gapBox.generateCanvasTexture()
}
```
If need negative margin, you need to create a shader or something to implement that
